File: array/Solution48.java
class Solution48 {
	public void rotate(int[][] matrix) {
		int n = matrix.length;
		int[][] copy = new int[n][n];
		for(int i = 0; i < n; i++){
			for(int j = 0; j < n; j++){
				copy[i][j] = matrix[i][j];
			}
		}
		int col = n - 1;
		for(int i = 0; i < n; i++){
			for(int j = 0; j < n; j++){
				matrix[j][col] = copy[i][j];
			}
			col -= 1;
		}
	}

    public static void main(String[] args){
    	Solution48 s = new Solution48();
    	int[][] test = new int[][]{{5, 1, 9, 11}, {2, 4, 8, 10}, {13, 3, 6, 7}, {15, 14, 12, 16}};
    	s.rotate(test);
    	for(int i = 0; i < 4; i++){
    		for(int j = 0; j < 4; j++){
    			System.out.print(test[i][j] + " ");
    		}
    		System.out.println();
    	}
    }
}

File: C++/Solution48.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

class Solution48 {
public:
    void rotate(vector<vector<int>>& matrix) {
        int size = matrix.size();
        vector<vector<int>> copy(matrix.begin(), matrix.end());
        for(int i = 0; i < size; i++){
            for(int j = 0; j < size; j++){
                matrix[i][j] = copy[size - 1 - j][i];
            }
        }
    }
};

int main(){
    vector<vector<int>> test = {{1, 2}, {2, 3}};
    vector<vector<int>> copy(test.begin(), test.end());
    test[0][1] = 20;
    cout << copy[0][1] << " " << test[0][1];
    return 0;
}